scilpy/tractograms/dps_and_dpp_management.py

```diff
@@ -1,12 +1,50 @@
 # -*- coding: utf-8 -*-
 import numpy as np
 
-from scilpy.viz.color import clip_and_normalize_data_for_cmap
+from nibabel.streamlines import ArraySequence
 
 
-def add_data_as_color_dpp(sft, cmap, data, clip_outliers=False, min_range=None,
-                          max_range=None, min_cmap=None, max_cmap=None,
-                          log=False, LUT=None):
+def get_data_as_arraysequence(data, ref_sft):
+    """ Get data in the same shape as a reference StatefulTractogram's
+    streamlines, so it can be used to set data_per_point or
+    data_per_streamline. The data may represent one value per streamline or one
+    value per point. The function will return an ArraySequence with the same
+    shape as the streamlines.
+
+    Parameters
+    ----------
+    data: np.ndarray
+        The data to convert to ArraySequence.
+    ref_sft: StatefulTractogram
+        The reference StatefulTractogram containing the streamlines.
+
+    Returns
+    -------
+    data_as_arraysequence: ArraySequence
+        The data as an ArraySequence.
+    """
+
+    if data.shape[0] == len(ref_sft):
+        data_as_arraysequence = ArraySequence(data)
+    elif data.shape[0] == ref_sft._get_point_count():
+        data_as_arraysequence = ArraySequence()
+        # This function was created to avoid messing with _data, _offsets and
+        # _lengths, so this feel kind of bad. However, the other way would be
+        # to create a new ArraySequence and iterate over the streamlines, but
+        # that would be way slower.
+        data_as_arraysequence._data = data
```

Comment:
Isn't there a way to hstack the data, or vstack, and then convert to array sequence?

In my PR #890 , I had done this. Is it really that much slower?

![image](https://github.com/user-attachments/assets/aebe4afc-0b03-4479-915d-7aa8dc52017c)

Reply:
Something I often do is that I create an array sequence with my streamlines like this:
dpp = sft.streamlines.copy()
dpp._data = my_array.copy()

That way everything is initialized with the right length, memory safe.

Reply:
@EmmaRenauld I feel like iterating over all streamlines would be problematic for large (>1M) tractograms. 
@frheault the problem with your approach is the dpp is still "prealocated" using the streamlines, which is weird. I think this is a deeper problem stemming from the way ArraySequences work but I'd rather not have to preallocate data per points as the streamlines' points.  Especially if more complicated processing is needed and some DPP may be left to the streamline's value by accident.

Reply:
@EmmaRenauld @frheault should be a bit cleaner now.